src/DotNet.Status.Web/DotNet.Status.Web/Controllers/GitHubHookController.cs

```diff
@@ -179,10 +182,176 @@ public async Task<IActionResult> IssuesHook()
+                    }
+                    break;
+                case "labeled":
+                case "reopened":
+                    // was the epic label applied? check to see if the milestone name already exists. If it doesn't, create a milestone with the issue name
+                    if ((action == "labeled" && issueEvent.Label.Name.Equals("Epic", StringComparison.OrdinalIgnoreCase)) || action == "reopened")
+                    {
+                        string epicName = issueEvent.Issue.Title;
+                        string epicUrl = issueEvent.Issue.HtmlUrl;
+
+                        // check to see if it currently exists and is open.
+                        Milestone foundMilestone = await GetMilestone(epicName);
+                        if (foundMilestone == null)
+                        {
+                            await CreateMilestoneAndLinkToIssue();
+                        }
+                        // Cannot create new milestones with the same name of one that exists even if it's closed, so we'll 
+                        //   re-open the closed one and update the description. If it already exists for some reason, we'll 
+                        //   assign the issue to the milestone, and link it in the description.
+                        else
+                        {
+                            // reopen the old one
+                            MilestoneUpdate update = new MilestoneUpdate
+                            {
+                                State = ItemState.Open,
+                                // add link to epic issue in the milestone description. 
+                                Description = $"{epicUrl}\r\n\r\nPrevious Description: {foundMilestone.Description}"
+                            };
+
+                            Milestone milestone = await client.Issue.Milestone.Update(org, repo, foundMilestone.Number, update);
+
+                            IssueUpdate issueUpdate = new IssueUpdate
+                            {
+                                Milestone = milestone.Number
+                            };
+
+                            await client.Issue.Update(org, repo, issueEvent.Issue.Number, issueUpdate);
+                        }
+                    }
+                    break;
+                case "unlabeled":
+                    // are we trying to remove the epic label? check to see if there are any open issues (not the current issue) in the milestone.
+                    if (issueEvent.Label.Name.Equals("Epic", StringComparison.OrdinalIgnoreCase) && issueEvent.Issue.Milestone?.OpenIssues > 1)
+                    {
+                        await client.Issue.Labels.AddToIssue(org, repo, issueEvent.Issue.Number, new string[] { "Epic" });
+                        _logger.LogInformation("Issue {0}/{1}#{2} is the epic of a milestone that currently contains open issues. Adding the Epic label back to this issue", org, repo, issueEvent.Issue.Number);
```

Comment:
```suggestion
                        _logger.LogInformation("Issue {Organization}/{Repository}#{IssueNumber} is the epic of a milestone that currently contains open issues. Adding the Epic label back to this issue", org, repo, issueEvent.Issue.Number);
```
App Insights uses the interpolation-like bits as queryable labels, so better to give them names matching what they represent.

Reply:
This seems like something that will confuse people if this rule is forgotten. 

If I remove the Epic label and the bot re-adds it, that will show up in the issue history, right? So that's good. We'll know approximately which automation is adding the label even if there's no exact context why. I wonder, can the bot add comments? If so, what do you think about it also adding a comment stating why it readded the label?

Reply:
> can the bot add comments? 

I think that it can. I'll have to check the permissions, but if it's not already set, I can update it. 

> If so, what do you think about it also adding a comment stating why it readded the label?

I thought about this, but worried that it might be too noisy? I'll solicit feedback from the team on this.

Reply:
Added this functionality :)